ClientLib/src/main/java/com/o3dr/android/client/utils/data/tlog/TLogParser.java

```diff
@@ -0,0 +1,349 @@
+     *
+     * @param uri
+     * @param filter
+     * @param callback
+     */
+    public static void getAllEventsAsync(final URI uri, final TLogParserFilter filter, final TLogParserCallback callback) {
+        parseTlogExecutor = Executors.newSingleThreadExecutor();
+        parseTlogExecutor.execute(new Runnable() {
+            @Override
+            public void run() {
+                File file = new File(uri.toString());
+                DataInputStream in = null;
+                LinkedList<Event> eventList = new LinkedList<>();
+                try {
+                    in = new DataInputStream(new BufferedInputStream(new FileInputStream(file)));
+                    while (in.available() > 0 && filter.continueIterating()) {
+                        Event event = next(in);
+                        if (filter.addEventToList(event)) {
+                            eventList.add(event);
+                        }
+                    }
+                    sendResult(callback, eventList);
+                } catch (EOFException e) {
+                    sendResult(callback, eventList);
+                } catch (FileNotFoundException e) {
+                    sendFailed(callback, e);
+                } catch (IOException e) {
+                    sendFailed(callback, e);
+                } finally {
+                    if (in != null) {
+                        try {
+                            in.close();
+                        } catch (IOException e) {
+                            Log.e(LOG_TAG, "Failed to close file " + uri, e);
+                        }
+                    }
+                }
+            }
+        });
+    }
+
+    private static void sendResult(final TLogParserCallback callback, final List<Event> events) {
+        if (callback != null) {
+            callback.onResult(events);
+        }
+    }
+
+    private static void sendFailed(final TLogParserCallback callback, final Exception e) {
+        if (callback != null) {
+            callback.onFailed(e);
+        }
+    }
+
+    private static Event next(DataInputStream in) throws IOException {
+        long timestamp = in.readLong() / 1000;
+        MAVLinkPacket packet;
+        while ((packet = parser.mavlink_parse_char(in.readUnsignedByte())) == null) ;
+        return new Event(timestamp, packet.unpack());
+    }
+
+    /**
+     * Mavlink message event.
+     */
+    public static class Event {
+        private long timestamp;
+        private MAVLinkMessage mavLinkMessage;
+
+        public Event(long timestamp, MAVLinkMessage mavLinkMessage) {
+            this.timestamp = timestamp;
+            this.mavLinkMessage = mavLinkMessage;
+        }
+
+        public long getTimestamp() {
+            return timestamp;
+        }
+
+        public MAVLinkMessage getMavLinkMessage() {
+            return mavLinkMessage;
+        }
+    }
+
+    /**
+     * Callback for asynchronous TLog iterator.
+     */
+    public interface TLogIteratorCallback {
```

Comment:
Do you want all the interfaces in once class file, or each in their own separate one